hazelcast/src/test/java/com/hazelcast/client/map/impl/nearcache/ClientMapNearCacheTest.java

```diff
@@ -287,6 +287,28 @@ public void run() {
         });
     }
 
+    @Test
+    public void testAfterDeleteAsyncNearCacheIsInvalidated() {
+        int mapSize = 1000;
+        String mapName = randomMapName();
+        hazelcastFactory.newHazelcastInstance(newConfig());
+        HazelcastInstance client = getClient(hazelcastFactory, newInvalidationOnChangeEnabledNearCacheConfig(mapName));
+
+        final IMap<Integer, Integer> clientMap = client.getMap(mapName);
+        populateMap(clientMap, mapSize);
+        populateNearCache(clientMap, mapSize);
+
+        for (int i = 0; i < mapSize; i++) {
+            clientMap.deleteAsync(i);
+        }
+
+        assertTrueEventually(new AssertTask() {
+            public void run() {
+                assertThatOwnedEntryCountEquals(clientMap, 0);
+            }
+        });
+    }
+
```

Comment:
Can you also please add tests for the server-side proxy? You can take a look at the tests that use removeAsync to see where to add tests to

Reply:
I looked but haven't found. There is MapProxyImplTest without any test cases for removeAsync. And there seems to be no test class for NearCachedMapProxyImpl.

Reply:
Could you please give me a more specific pointer, where to look?

Reply:
When I looked at the usages of IMap#removeAsync, I see the tests. (I am using Intellij Idea). 

Here is the screenshot of the usage, you can check these tests and write equivalent tests if needed for the deleteAsync
<img width="642" alt="image" src="https://user-images.githubusercontent.com/29387993/216550567-f35053d3-3424-4941-9581-19a6f3ce09ec.png">

Reply:
OK, I will work on adding more tests next week

Reply:
com.hazelcast.client.map
* ClientMapBasicTest - added testDeleteAsync, testDeleteAsync_whenKeyNotPresent
* ClientMapTest - added testAsyncDelete
* MapPreconditionsTest - added testDeleteAsync


com.hazelcast.client.map.impl.nearcache
* ClientMapNearCacheTest - added testAfterDeleteAsyncNearCacheIsInvalidated

com.hazelcast.core.standalone
* AllTest - added deleteAsync operation to loadMapOperations()

com.hazelcast.map
* AsyncTest - added testDeleteAsync, testDeleteAsyncWithImmediateTimeout, testDeleteAsyncWithNonExistentKey
* BasicMapTest - added testGetPutDeleteAsync, extended testNullChecks
* LocalMapStatsTest - added testDeleteAsync
* MapPreconditionsTest - added testDeleteAsync

com.hazelcast.map.impl.mapstore
* EntryLoaderSimpleTest - added testDeleteAsync_returnsNull
* EntryStoreSimpleTest - added testDeleteAsync

com.hazelcast.map.impl.nearcache
* AbstractMapNearCacheLocalInvalidationTest - added testDeleteAsync

com.hazelcast.splitbrainprotection.map
* MapSplitBrainProtectionWriteTest - added deleteAsync_successful_whenSplitBrainProtectionSize_met, deleteAsync_failing_whenSplitBrainProtectionSize_met

Reply:
Didn't add anything to DistributedDatastructuresMetricsTest and EventQueuePluginTest in com.hazelcast.internal.diagnostics. removeAsync is used in a long test scenario checking other subsystems. Not sure if it's appropriate to duplicate the whole test scenario to use deleteAsync instead of removeAsync.

Reply:
Thanks a lot, I also agree that we don't have to duplicate the tests for the two you wrote